File: backend/src/sor/models/conflict.py
from __future__ import annotations

from pydantic import BaseModel, Field


class AgentPosition(BaseModel):
    agent_name: str
    position: str
    evidence: str = ""
    confidence: float = 0.0


class AgreementPoint(BaseModel):
    topic: str
    summary: str
    supporting_agents: list[str] = Field(default_factory=list)
    evidence: list[str] = Field(default_factory=list)


class DisagreementPoint(BaseModel):
    topic: str
    summary: str
    positions: list[AgentPosition] = Field(default_factory=list)


class ConflictReport(BaseModel):
    stage: int
    agreements: list[AgreementPoint] = Field(default_factory=list)
    disagreements: list[DisagreementPoint] = Field(default_factory=list)
    unresolved_tensions: list[str] = Field(default_factory=list)
    synthesis: str = ""

File: ios/SocietyOfResearchers/Models/Conflict.swift
import Foundation

struct AgentPosition: Codable, Sendable {
    let agentName: String
    let position: String
    var evidence: String = ""
    var confidence: Double = 0.0

    enum CodingKeys: String, CodingKey {
        case position, evidence, confidence
        case agentName = "agent_name"
    }
}

struct AgreementPoint: Codable, Identifiable, Sendable {
    let topic: String
    let summary: String
    var supportingAgents: [String] = []
    var evidence: [String] = []

    var id: String { topic }

    enum CodingKeys: String, CodingKey {
        case topic, summary, evidence
        case supportingAgents = "supporting_agents"
    }
}

struct DisagreementPoint: Codable, Identifiable, Sendable {
    let topic: String
    let summary: String
    var positions: [AgentPosition] = []

    var id: String { topic }
}

struct ConflictReport: Codable, Sendable {
    let stage: Int
    var agreements: [AgreementPoint] = []
    var disagreements: [DisagreementPoint] = []
    var unresolvedTensions: [String] = []
    var synthesis: String = ""

    enum CodingKeys: String, CodingKey {
        case stage, agreements, disagreements, synthesis
        case unresolvedTensions = "unresolved_tensions"
    }
}
